src/hacking/building-for-openharmony.md

```diff
@@ -105,15 +110,84 @@ Depending on your Linux distributions you may want to use a different group.
 To check if `hdc` is now working, you can run `hdc list targets` and it should show your device serial number.
 If it doesn't work, try rebooting.
 
-Please note, that your phone needs to be in "Developer mode" with USB debugging enabled.
+Please note, that your device needs to be in "Developer mode" with USB debugging enabled.
 The process here is exactly the same as one android:
 1. Tap the build number multiple times to enable developer mode.
 2. Then navigate to the developer options and enable USB debugging.
 3. When you connect your device for the first time, confirm the pop-up asking you if you want to trust the computer you are connecting to.
 
+## Signing configuration
+
+Most devices require that the HAP is digitally signed by the developer to be able to install it.
+When using the `hvigor` tool, this can be accomplished by setting a static `signingConfigs` object in the `build-profile.json5` file or
+by dynamically creating the `signingConfigs` array on the appication context object using the `hvigorfile.ts` script.
+
+The `signingConfigs` property is an array of objects with the following structure:
+
+```json
+
+{
+    "name": "default",
+    "type": "OpenHarmony",
+    "material": {
+        "certpath": "/path/to/app-signing-certificate.cer",
+        "storePassword": "<encrypted password>",
+        "keyAlias": "debugKey",
+        "keyPassword": "<encrypted password>",
+        "profile": "/path/to/signed-profile-certificate.p7b",
+        "signAlg": "SHA256withECDSA",
+        "storeFile": "/path/to/java-keystore-file.p12"
+    }
+}
+```
+
+Here <encrypted password> is a hexadecimal string representation of the plaintext password after being encrypted.
+The key and salt information used to encrypt these passwords are generated by DevEco Studio IDE and are stored on-disk alongside
+the certificate files and keystore, usually under `<USER HOME>/.ohos/config/openharmony`.
+
+You can use the IDE to generate information needed for password encryption, the required application and
```

Comment:
Probably we should mention that this is only for developing / testing purposes. To ship an app, one would need to get the signature configuration from the app store provider.

Reply:
done in [4090da6](https://github.com/servo/book/pull/24/commits/4090da6a8f2698c7c1b68402252580a44c5187bd)